plugin/kubernetes/object/endpoint.go

```diff
@@ -1,18 +1,23 @@
 package object
 
 import (
+	"time"
+
+	"github.com/coredns/coredns/plugin/pkg/log"
+
 	api "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
 // Endpoints is a stripped down api.Endpoints with only the items we need for CoreDNS.
 type Endpoints struct {
-	Version   string
-	Name      string
-	Namespace string
-	Index     string
-	IndexIP   []string
-	Subsets   []EndpointSubset
+	Version               string
+	Name                  string
+	Namespace             string
+	Index                 string
+	IndexIP               []string
+	Subsets               []EndpointSubset
+	LastChangeTriggerTime time.Time
```

Comment:
this increases memory used

Reply:
I obviously don't know this code well enough, but I think this is acceptable.

In general, we are currently saying that we support up to 10k services in k8s. We are planning to bump it in the future to 100k. but given time.Time is 8B IIRC, we are talking about 80kB now and 800kB in the future.

I wouldn't consider it as a problem (epsecially given that a single endpoints object may have 1mB size...)